### DISCUSSION TITLE
voltronic AXPERT KS

### DISCUSSION THREAD

**Discussion by G14nLu** [jblance/mpp-solar#49] [Q&A] [answered]
Hi i'm trying to use your repo with voltronic AXPERT KS, but i'm facing some errors.
I'm sure of hardware connection because i've tested it in many different ways.
The command is
 mpp-solar -n 'FirstInverter' -P PI16 -o screen -c QPIGS -p /dev/ttyUSB0
The error is
 log.info(f"attempting to create output processor: {output}")

Same errore with PI00 protocol.

Any advice? many thanks

**Comment by jblance** [comment, marked as answer]
At a guess its because your python version is not new enough. It needs to be 3.6 or later

You can get more info from running the command with debug enabled, I.e. add the ```-D``` option to the command

**Reply by G14nLu** [reply]
Many thanks for your answer. I tried the venv method following the instructions, just to be sure to not have python version issues.
everything works ok till the command mpp-solar -v.
the complete command 
`mpp-solar -n 'FirstInverter' -P PI00 -o screen -c QPIGS -p /dev/USB0`
doesn't work, giving the error reported in the previous message. any advice?

I'm working on raspberry pi 3b+

thank you

**Reply by G14nLu** [reply]
I've tried with a clean raspberry os install and venv method. this is the result
![image](https://user-images.githubusercontent.com/17130459/104327635-49f87700-54eb-11eb-9b5c-834039f32c85.png)

**Reply by jblance** [reply]
Bugger, thats an error that occurs if the ble python module is missing - have removed that for now
If you update to the latest version it should work
As always if you are not getting results, adding `-D` which is the debug flag to the command give more info to help find the problem

**Reply by G14nLu** [reply]
installed `sudo apt-get install libglib2.0-dev`
then retried. but...

![image](https://user-images.githubusercontent.com/17130459/104440580-d5791300-5592-11eb-8201-e6f576b0783d.png)

**Reply by jblance** [reply]
What version are you running?

**Reply by G14nLu** [reply]
finally it worked. installing from source and selecting PI30 as protocol.
Many thanks
